service/snap/snap.go

```diff
@@ -359,28 +303,16 @@ func (s Service) Installed() (bool, error) {
 //
 // InstallCommands is part of the service.Service interface
 func (s Service) InstallCommands() ([]string, error) {
-	commands := make([]string, 0, 1+len(s.app.Prerequisites))
-	confinementPolicy := confimentParameterAsString(s.app.ConfinementPolicy)
-
-	for _, prerequisite := range s.app.Prerequisites {
-		command := fmt.Sprintf("%v install --channel=%v %v %v",
-			s.executable,
-			prerequisite.Channel,
-			confinementPolicy,
-			prerequisite.Name,
-		)
-		logger.Infof("preparing command: %v", command)
-		commands = append(commands, command)
-	}
-
-	command := fmt.Sprintf("%v install --channel=%v %v %v",
-		s.executable,
-		s.app.Channel,
-		confinementPolicy,
-		s.app.Name,
-	)
-	logger.Infof("preparing command: %v", command)
-	commands = append(commands, command)
+	deps := s.app.Prerequisites()
+	commands := make([]string, 1+len(deps))
```

Comment:
You could always change this to `commands := make([]string, 0, 1+len(deps))` and just append ;-)

Reply:
Actually, you are using the append pattern below.